### DISCUSSION TITLE
How to build libllama.so with the new cmake

### DISCUSSION THREAD

**Discussion by marcov-dart** [ggml-org/llama.cpp#10938] [Q&A] [answered]
I used to be able to do make -j libllama.so to get libllama.so and libggml.so. But I don't know how to do it now with the new cmake setup. Does anyone here know?

**Comment by marcov-dart** [comment]
Oh lol, removing the first 3 lines from the Makefile and then doing:

> make -j libllama.so

Still works... I did not expect that.

**Reply by danbev** [reply, marked as answer]
I should be possible to use the following command to build with `cmake`:
```console
$ cmake --build build --target llama
```
And the shared library should be created in `build/src/` afterwards.

**Comment by marcov-dart** [comment]
Thanks for your answer. I tried it:

[ 23%] Built target ggml-base
[ 35%] Built target ggml-metal
[ 64%] Built target ggml-cpu
[ 70%] Built target ggml-blas
[ 76%] Built target ggml
[100%] Built target llama

But no .so files in build/src/

cd build/src 
(base) marco@MAVs-iMac src % ls -al
total 2464
drwxr-xr-x   6 marco  staff      192 Dec 23 10:20 .
drwxr-xr-x  21 marco  staff      672 Dec 23 10:18 ..
drwxr-xr-x   5 marco  staff      160 Dec 23 10:18 CMakeFiles
-rw-r--r--   1 marco  staff    13136 Dec 23 10:18 Makefile
-rw-r--r--   1 marco  staff     1195 Dec 23 10:18 cmake_install.cmake
-rwxr-xr-x   1 marco  staff  1237408 Dec 23 10:20 libllama.dylib

**Reply by danbev** [reply]
On mac this would generate a Mach-O-based shared library with the `.dylb` extension, and on Linux it would generate an ELF-based shared object with the `.so` extension. In this case the shared library is in:
```console
 $ file  build/src/libllama.dylib
build/src/libllama.dylib: Mach-O 64-bit dynamically linked shared library arm64
```

**Comment by marcov-dart** [comment]
Thanks so much!

I used the files build/src/libllama.dylib and build/ggml/src/libggml.dylib to replace my old .so files and I can confirm it works.
